File: 프로그래머스/lv2/42578. 의상/의상.py
# def solution(clothes):
#     answer = 1
#     #서로다른 옷의 조합 수 리턴, [의상이름,의상종류] 같은이름존재x 길이 20이하 알파벳소문자 혹은_로만 이루어짐 의상수는 1~30, 하루 최소 한 개의상입음 
#     dict={}
#     for i in clothes:
#         #if !dict[i[1]] :
#         if i[1] not in dict.keys() :
            
#             dict[i[1]]=[]
#             dict[i[1]].append(i[0])

#             continue
#         else :
#             dict[i[1]].append(i[0])
#             continue
    
#     print (dict)
#     keys=dict.keys()
#     for i in keys:
#         answer=answer*(len(dict[i])+1)
        
    
#     return answer-1

def solution(clothes):
    dict={}
    for i in clothes:
        if i[1] not in dict.keys():
            dict[i[1]]=[i[0]]
        else :
            dict[i[1]].append(i[0])
    answer=1
    print(dict)
    for j in dict.values():
        answer*= len(j)+1
    return answer-1 #아예 안입는건 안됨 
        

File: 프로그래머스/2/42578. 의상/의상.js
//clotes=[['아이템',아이템 분류'],[...], ...]

function solution(clothes) {
    let answer = 1;
    let obj={}
    for ( i of clothes){    //clothes 순회하면서 1번인덱스 기준으로 object에 정리 
        // if (Object.keys(obj).includes(i[1])){    
        // if (i[1] in obj){
        if (obj.hasOwnProperty(i[1])){
            obj[i[1]].push(i[0])
        }else{
            obj[i[1]]=[i[0]]
        }
    }
    for (key in obj){ //key를 기준으로 아이템 착용 경우의 수대로 곱해준다.
        answer=answer*(obj[key].length+1)
    }
    return answer-1;// 아무것도 착용하지 않은경우 제외 
}

// function solution(clothes){
//     let result=0
//     //2차원 배열 clothes주어지고, 서로 다른 옷의 조합의 수를 return 
//     let clothesObj={}
//     // clothes.reduce((el)=>{
//     //     clothesObj[el[1]]=clothesObj[el[1]]? [...clothesObj[el[1]],el[0]] : [el[0]]
//     // })
//     for (let i =0 ; i<clothes.length; i++){
//         if (clothesObj[clothes[i][1]]){
//             clothesObj[clothes[i][1]].append(clothes[i][0])
//         }else { clothesObj[clothes[i][1]]=[clothes[i][0]]}
//     }
//     console.log(clothesObj)
    
//     return result
// }